tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py

```diff
@@ -0,0 +1,322 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import json
+import tensorflow as tf
+from IPython import display
+from google.colab import output
+from google.protobuf import json_format
+from tensorboard.plugins.interactive_inference.utils import inference_utils
+
+# Python functions for requests from javascript.
+def infer_examples():
+  WitWidget.widget.infer()
+output.register_callback('notebook.InferExamples', infer_examples)
+
+def delete_example(index):
+  WitWidget.widget.delete_example(index)
+output.register_callback('notebook.DeleteExample', delete_example)
+
+def duplicate_example(index):
+  WitWidget.widget.duplicate_example(index)
+output.register_callback('notebook.DuplicateExample', duplicate_example)
+
+def update_example(index, example):
+  WitWidget.widget.update_example(index, example)
+output.register_callback('notebook.UpdateExample', update_example)
+
+def get_eligible_features():
+  WitWidget.widget.get_eligible_features()
+output.register_callback('notebook.GetEligibleFeatures', get_eligible_features)
+
+def infer_mutants(details):
+  WitWidget.widget.infer_mutants(details)
+output.register_callback('notebook.InferMutants', infer_mutants)
+
+# HTML/javascript for the WIT frontend.
+WIT_HTML = """
+  <tf-interactive-inference-dashboard id="wit" local>
+  </tf-interactive-inference-dashboard>
+  <script>
+    const examples = {examples};
+    const wit = document.querySelector("#wit");
```

Comment:
Do note that this code won't be vulcanized/transpiled so it might be better to write it in es5 fashion depending on your desired supported browsers.